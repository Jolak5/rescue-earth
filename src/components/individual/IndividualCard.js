import React from 'react';
import { NavLink } from 'react-router-dom';
import { useSelector } from 'react-redux';
import './individualCard.css';
import { Back } from '../icon';

export default function IndividualCard() {
  const forex = useSelector((state) => state.Forex.forexItem);

  return (
    <div className="modal">
      <header className="header">
        <NavLink to="/"><Back /></NavLink>
        <h1>Global Covid 19 Stats</h1>
      </header>
      <div className="global individual">
        <div className="global-details stats">
          {forex.map((item) => (
            <article className="article" key={item.id}>
              <div>
                <h1>
                  {item.id}
                </h1>
                <p>
                  Confirmed cases:
                  <p>{item.Deaths}</p>

                </p>
              </div>
            </article>
          ))}
        </div>
      </div>
      {forex.map((item) => (

        <div key={item.Death}>
          <ul className="individual-item">
            <li>
              Confirmed cases
              {' '}
              <p>{item.confirmed}</p>
            </li>
            <li>
              Total deaths
              {' '}
              <p>{item.Deaths}</p>
            </li>
            <li>
              Last_Updated
              {' '}
              <p>{item.Last_Update}</p>
            </li>
            <li>
              Fatality rate:
              {' '}
              <p>{item.Fatality}</p>

            </li>

          </ul>
        </div>
      ))}
    </div>
  );
}
