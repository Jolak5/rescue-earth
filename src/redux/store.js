import { configureStore } from '@reduxjs/toolkit';
import pollutionReducer from './homepage/homepageSlice';

export default configureStore({
  reducer: {
    pollution: pollutionReducer,
  },
});
