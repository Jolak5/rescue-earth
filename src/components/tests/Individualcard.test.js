import React from 'react';
import { render } from '@testing-library/react';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/store';

import IndividualCard from '../individual/IndividualCard';

// Test case for the dynamic component
describe('IndividualCard', () => {
  it('renders IndividualCard data with NavLink components', async () => {
    const homeLink = render(
      <Router>
        <Provider store={store}>
          <IndividualCard />
        </Provider>
      </Router>,
    );
    expect(homeLink).toMatchSnapshot();
  });
});
