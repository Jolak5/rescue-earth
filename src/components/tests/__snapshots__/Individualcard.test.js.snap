// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders Homepage data with NavLink components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal"
      >
        <header
          class="header"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <svg
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              />
            </svg>
          </a>
          <h1>
            Global Covid 19 Stats
          </h1>
        </header>
        <div
          class="global individual"
        >
          <div
            class="global-details stats"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal"
    >
      <header
        class="header"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <svg
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            />
          </svg>
        </a>
        <h1>
          Global Covid 19 Stats
        </h1>
      </header>
      <div
        class="global individual"
      >
        <div
          class="global-details stats"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`IndividualCard renders IndividualCard data with NavLink components 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal"
      >
        <header
          class="header"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <svg
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              />
            </svg>
          </a>
          <h1>
            Global Covid 19 Stats
          </h1>
        </header>
        <div
          class="global individual"
        >
          <div
            class="global-details stats"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal"
    >
      <header
        class="header"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <svg
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            />
          </svg>
        </a>
        <h1>
          Global Covid 19 Stats
        </h1>
      </header>
      <div
        class="global individual"
      >
        <div
          class="global-details stats"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
